configs/projects/pe-installer-runtime.rb

```diff
@@ -0,0 +1,142 @@
+  # tweak or adjust them as needed.
+  proj.setting(:cppflags, "-I#{proj.includedir} -I/opt/pl-build-tools/include")
+  proj.setting(:cflags, "#{proj.cppflags}")
+  proj.setting(:ldflags, "-L#{proj.libdir} -L/opt/pl-build-tools/lib -Wl,-rpath=#{proj.libdir}")
+
+  # Platform specific overrides or settings, which may override the defaults
+  if platform.is_windows?
+    arch = platform.architecture == "x64" ? "64" : "32"
+    proj.setting(:gcc_root, "C:/tools/mingw#{arch}")
+    proj.setting(:gcc_bindir, "#{proj.gcc_root}/bin")
+    proj.setting(:tools_root, "C:/tools/pl-build-tools")
+    proj.setting(:cppflags, "-I#{proj.tools_root}/include -I#{proj.gcc_root}/include -I#{proj.includedir}")
+    proj.setting(:cflags, "#{proj.cppflags}")
+    proj.setting(:ldflags, "-L#{proj.tools_root}/lib -L#{proj.gcc_root}/lib -L#{proj.libdir} -Wl,--nxcompat -Wl,--dynamicbase")
+    proj.setting(:cygwin, "nodosfilewarning winsymlinks:native")
+  end
+
+  if platform.is_macos?
+    # For OS X, we should optimize for an older architecture than Apple
+    # currently ships for; there's a lot of older xeon chips based on
+    # that architecture still in use throughout the Mac ecosystem.
+    # Additionally, OS X doesn't use RPATH for linking. We shouldn't
+    # define it or try to force it in the linker, because this might
+    # break gcc or clang if they try to use the RPATH values we forced.
+    proj.setting(:cppflags, "-I#{proj.includedir}")
+    proj.setting(:cflags, "-march=core2 -msse4 #{proj.cppflags}")
+    proj.setting(:ldflags, "-L#{proj.libdir} ")
+  end
+
+  # These flags are applied in addition to the defaults in configs/component/openssl.rb.
+  proj.setting(:openssl_extra_configure_flags, [
+    'no-dtls',
+    'no-dtls1',
+    'no-idea',
+    'no-seed',
+    'no-ssl2-method',
+    'no-weak-ssl-ciphers',
+    '-DOPENSSL_NO_HEARTBEATS',
+  ])
+
+  # What to build?
+  # --------------
+
+  # Ruby and deps
+  proj.component "openssl"
+  proj.component "runtime-pe-installer"
+  proj.component "puppet-ca-bundle"
+  proj.component "ruby-#{proj.ruby_version}"
+
+  # Puppet dependencies
+  proj.component 'rubygem-deep-merge'
+  proj.component 'rubygem-text'
+  proj.component 'rubygem-locale'
+  proj.component 'rubygem-gettext'
+  proj.component 'rubygem-fast_gettext'
+  proj.component 'rubygem-semantic_puppet'
+
+  # R10k dependencies
+  proj.component 'rubygem-gettext-setup'
+
+  # Core dependencies
+  proj.component 'rubygem-ffi'
+  proj.component 'rubygem-minitar'
+  proj.component 'rubygem-multi_json'
+  proj.component 'rubygem-net-ssh'
+
+  # Core Windows dependencies
+  proj.component 'rubygem-win32-dir'
+  proj.component 'rubygem-win32-process'
+  proj.component 'rubygem-win32-security'
+  proj.component 'rubygem-win32-service'
+
+  # What to include in package?
+  proj.directory proj.prefix
+
+  # Export the settings for the current project and platform as yaml during builds
+#  proj.publish_yaml_settings
```

Comment:
This line allows downstream projects that consume the tarballs produced by puppet-runtime to [load their settings out of the yaml file](https://github.com/puppetlabs/puppet-agent/blob/master/configs/projects/puppet-agent.rb#L15) instead of cloning the entire runtime project to gather the settings that way. You'll may want to uncomment this line even if you don't end up doing inheritance the way puppet-agent does it, since it can be helpful to see the settings for an existing build without having to use vanagon's `inspect` command on the same revision.

Reply:
Ah, ok. I was just noticing that it errored out when I tried it. I'll try again and post the output here.

Reply:
Ah, @highb one more thing to note about this line - it requires vanagon >= 0.15.12, so if your local version is older than that, it will break the build

Reply:
A `bundle update` fixed it completely. Thanks!